src/components/Inventory/view/ChangeQuantityModal.tsx

```diff
@@ -0,0 +1,127 @@
+import useSnackbar from 'src/utils/useSnackbar';
+import useForm from 'src/utils/useForm';
+import stopPropagation from 'src/utils/stopPropagationEvent';
+
+export interface ChangeQuantityModalProps {
+  open: boolean;
+  onClose: () => void;
+  modalValues: ModalValuesType;
+  row: Batch;
+  reload?: () => void;
+}
+
+export default function ChangeQuantityModal(props: ChangeQuantityModalProps): JSX.Element {
+  const { onClose, open, modalValues, row, reload } = props;
+  const { type } = modalValues;
+  const [movedValue, setMovedValue] = useState<number | undefined>();
+  const [record, setRecord] = useForm(row);
+  const snackbar = useSnackbar();
+
+  const onSubmit = async () => {
+    const response = await updateBatchQuantities({ ...record, version: row.version });
+    if (response.requestSucceeded) {
+      if (reload) {
+        reload();
+      }
+      onClose();
+    } else {
+      snackbar.toastError(response.error);
+    }
+  };
+
+  const onChangeMovedValue = (value: unknown) => {
+    if (value && !isNaN(value as number) && (value as number) > 0) {
+      const valueNumber = value as number;
+      setMovedValue(valueNumber);
+      if (type === 'germinating') {
+        setRecord({
+          ...row,
+          germinatingQuantity: +row.germinatingQuantity - valueNumber,
+          notReadyQuantity: +row.notReadyQuantity + +valueNumber,
+        });
+      } else {
+        setRecord({
+          ...row,
+          notReadyQuantity: +row.notReadyQuantity - valueNumber,
+          readyQuantity: +row.readyQuantity + +valueNumber,
+        });
+      }
+    } else {
+      setRecord(row);
+      setMovedValue(undefined);
+    }
+  };
+
+  const onCloseHandler = () => {
+    setMovedValue(undefined);
+    setRecord(row);
+    onClose();
+  };
+
+  const handleClickAway = (event?: any) => {
+    stopPropagation(event);
+  };
+
+  return (
+    <div onClick={handleClickAway}>
+      <DialogBox
+        onClose={onCloseHandler}
+        open={open}
+        title={type === 'germinating' ? strings.CHANGE_GERMINATING_STATUS : strings.CHANGE_NOT_READY_STATUS}
+        size='medium'
+        middleButtons={[
+          <Button
+            label={strings.CANCEL}
+            priority='secondary'
+            type='passive'
+            onClick={onCloseHandler}
+            size='medium'
+            key='button-1'
+          />,
+          <Button label={strings.SAVE} type='productive' onClick={onSubmit} size='medium' key='button-2' />,
+        ]}
+      >
+        <Grid>
+          <Grid item xs={10} textAlign='left' display='flex'>
+            <Textfield
+              value={type === 'germinating' ? record.germinatingQuantity : record.notReadyQuantity}
+              display={true}
+              label={type === 'germinating' ? strings.GERMINATING : strings.NOT_READY}
+              id={'previousValue'}
+              type={'number'}
+            />
+
+            <Box maxWidth='60px'>
```

Comment:
missed one thing, there's a bit more space between the labels in the design (looks a bit crowded in vercel)

<img width="194" alt="Nursery Management (Phase 2) 2022-11-02 09-36-55" src="https://user-images.githubusercontent.com/1865174/199551144-03661684-117c-44f9-b9ab-cd29e29b112f.png">